thoth/adviser/sieves/abi_compat.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+# thoth-adviser
+# Copyright(C) 2019, 2020 Kevin Postlehtwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Filter out stacks which have buildtime errors."""
+
+import logging
+
+import logging
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Generator
+from typing import TYPE_CHECKING
+
+import attr
+from thoth.python import PackageVersion
+from thoth.storages.exceptions import NotFoundError
+
+from ..sieve import Sieve
+
+if TYPE_CHECKING:
+    from ..pipeline_builder import PipelineBuilderContext
+
+_LOGGER = logging.getLogger(__name__)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AbiCompatability(Step):
+    """Remove packages if the image being used doesn't have necessary ABI."""
+
```

Comment:
There is missing `should_include` implementation - when this step pipeline unit should be included to resolver [1].

[1] https://thoth-station.ninja/docs/developers/adviser/unit.html#registering-a-pipeline-unit-to-pipeline

Reply:
Also, please include this pipeline step only if the environment is fully specified. Otherwise, it can end up passing `None` values to database queries:

```python
if runtime_environment.is_fully_specified():
    # ...
```